### ISSUE TITLE
Replacing single quote by double quote even though `singleQuote`: true

### ISSUE THREAD

**Issue by Vadorequest** [prettier/prettier#8459]
**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAxCF4BOSmASnAIZgwB02ArlFQJbQDCEAtgA7QIwA8ABQIQuaAHyYAvJgAUASmmTgAHSiZMkKBkzBMWAL7TMdNHAAqBctoA25GCygKA3Go0E4MOgXWy3GzH4AEyYAN38NcQiA4BhZAHIAMzxCakIONGobJigAa3iAGkx49igQh3RMAHMAS6gaqxs4NEwglRV4ugdstHtHePkDaP4AehDQqPVMeVcoA2cQApBRCu1kUHICEQB3QU2ENGQQchtt8gBPQ6WAIyswXM8AZS5KHKrkGAI6OCWACxgODYAOq-JjwNAvMBwR4HMFhMHnI5gNBXEA5MwEGDCchVDjkZCJE5mJYAKzQAA8AEJ3B4wR7kDhwAAyOTgBKJPxAZPJjzeTQAinR8GykISbMSQC8CBijtdyNc4DZFpKCDkYECmEEYL9kAAOAAMSy4IjMQKsXCOxuacAIoTZSwAjkL4MJRIcUOQ0ABaKBwOBBf3KjxOpgebG4-GijlLMwcJgfL6ctB8uCC4UJ75LGDyjVanVIABMWasTGyUCq7A4eKOzQArMrTBZ5e6xRLQt8AJJlPiPMCqrgwACCZUeMHOTXZ4rgBgMQA)

```sh
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 80,
  "proseWrap": "preserve",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false
}
```

**Input:**

```jsx
const Footer: React.FunctionComponent<Props> = () => {
  const { t } = useTranslation();
  return (
    <div
    >
      {t('footer.terms.link', 'Conditions générales d\'utilisation')}
    </div>
  );
};
```

**Output:**

```jsx
const Footer: React.FunctionComponent<Props> = () => {
  const { t } = useTranslation();
  return (
    <div
    >
      {t('footer.terms.link', "Conditions générales d'utilisation")}
    </div>
  );
};
```

**Expected behavior:**
It should leave `'Conditions générales d\'utilisation'` as it, it doesn't. This creates an error with my eslint rule.

**Comment by kachkaev** [comment]
👋 @Vadorequest, this is expected behaviour. See https://prettier.io/docs/en/options.html#quotes

> If the number of quotes outweighs the other quote, the quote which is less used will be used to format the string - Example: `"I'm double quoted"` results in `"I'm double quoted"` and `"This \"example\" is single quoted"` results in `'This "example" is single quoted'`.

**Comment by Vadorequest** [comment]
Thanks, the underlying issue was that it broke my eslint rule and I thought it wasn't supposed to be the case. (eslint force to single, while prettier force double)

Had to change the eslint configuration to handle that edge case.

**Comment by kachkaev** [comment]
Try extending this config in your `.eslintrc`: https://github.com/prettier/eslint-config-prettier

When added _after_ all configs,  `eslint-config-prettier` guarantees no rule clash 👍

**Comment by Vadorequest** [comment]
Using the following .eslintrc:

```yml
---
env:
  browser: true
  commonjs: true
  es6: true
  node: true
extends:
  - plugin:@typescript-eslint/recommended
  - plugin:react/recommended
  - plugin:jsx-a11y/recommended
  - plugin:prettier/recommended
  - prettier/react
  - prettier/@typescript-eslint
  - prettier # Should be last - https://github.com/prettier/eslint-config-prettier
globals:
  Atomics: readonly
  SharedArrayBuffer: readonly
plugins:
  - jest
  - react
  - react-hooks
  - prettier
parser: '@typescript-eslint/parser'
parserOptions:
  project: ./tsconfig.json
settings:
  react:
    version: detect
rules: # See https://eslint.org/docs/rules
  semi:
    - error
    - always # Always put commas, to avoid multilines git diff when new lines are added
  quotes:
    - error
    - single # Prefer simple quotes
    - allowTemplateLiterals: true # Allow the use of `` instead of '' and don't try to replace it, even when `` isn't needed
  comma-spacing:
    - error
    - before: false
      after: true
  indent:
    - error
    - 2
    - SwitchCase: 1
  arrow-parens:
    - error
    - always
  max-len: 0 # Disable line length checks, because the IDE is already configured to warn about it, and it's a waste of time to check for lines that are too long, especially in comments (like this one!)
  strict: 'off'
  no-console: 1 # Shouldn't use "console", but "logger" instead
  allowArrowFunctions: 0
  no-unused-vars: 0 # Disabled, already handled by @typescript-eslint/no-unused-vars
  import/prefer-default-export: 0 # When there is only a single export from a module, don't enforce a default export, but rather let developer choose what's best
  no-else-return: 0 # Don't enforce, let developer choose. Sometimes we like to specifically use "return" for the sake of comprehensibility and avoid ambiguity
  no-underscore-dangle: 0 # Allow _ before/after variables and functions, convention for something meant to be "private"
  arrow-body-style: 0 # Don't enforce, let developer choose. Sometimes we like to specifically use "return" for ease of debugging and printing
  quote-props:
    - warn
    - consistent-as-needed # Enforce consistency with quotes on props, either all must be quoted, or all unquoted for a given object
  no-return-await: 0 # Useful before, but recent node.js enhancements make it useless on node 12+ (we use 10, but still, for consistency) - Read https://stackoverflow.com/questions/44806135/why-no-return-await-vs-const-x-await
  no-extra-boolean-cast: 0 # Don't enforce, let developer choose. Using "!!!" is sometimes useful (edge cases), and has a semantic value (dev intention)
  object-curly-newline:
    - warn
    - ObjectExpression:
        multiline: true
        minProperties: 5
        consistent: true
      ObjectPattern:
        multiline: true
        minProperties: 5
        consistent: true
      ImportDeclaration:
        multiline: true
        minProperties: 8 # Doesn't play so well with webstorm, which wraps based on the number of chars in the row, not based on the number of props #sucks
        consistent: true
      ExportDeclaration:
        multiline: true
        minProperties: 5
        consistent: true
  prettier/prettier: error
  react-hooks/rules-of-hooks: error
  react-hooks/exhaustive-deps: warn
  react/jsx-no-target-blank: warn # Not using "noreferrer" is not a security risk, but "noopener" should always be used indeed
  react/prop-types: warn # Should be handled with TS instead
  react/no-unescaped-entities: 0 # Causes text mismatch when enabled
  jsx-a11y/anchor-is-valid: warn
  linebreak-style:
    - error
    - unix
  '@typescript-eslint/ban-ts-ignore': warn # ts-ignore are sometimes the only way to bypass a TS issue, we trust we will use them for good and not abuse them
  '@typescript-eslint/no-use-before-define': warn
  '@typescript-eslint/no-unused-vars':
    - warn
    -
      vars: 'all' # We don't want unused variables (noise) - XXX Note that this will be a duplicate of "no-unused-vars" rule
      args: 'none' # Sometimes it's useful to have unused arguments for later use, such as describing what args are available (DX)
      ignoreRestSiblings: true # Sometimes it's useful to have unused props for later use, such as describing what props are available (DX)
overrides:
  - files: ['**/*.tsx']
    rules:
      'react/prop-types': 'off'

```

I still get 
![image](https://user-images.githubusercontent.com/3807458/83360154-6cd08380-a37f-11ea-9c18-4178b62ceda4.png)

**Comment by kachkaev** [comment]
This could be because you are also using prettier plugin, not just as config. Also it could be a matter of VSCode not being in sync with your config updates.

Feel free to reuse my config for inspiration – it does not suffer from this issue: https://github.com/kachkaev/routine-npm-packages/blob/master/packages/eslint-config-base/index.js

**Comment by Vadorequest** [comment]
Thanks! 

I'll take a look, but I'm using WebStorm, not VSCode. (WebStorm has a native prettier plugin, but I'm not sure how it works)

**Comment by thorn0** [comment]
Works as expected on Prettier's side.

**Comment by thorn0** [comment]
@Vadorequest BTW, [this tool](https://github.com/prettier/eslint-config-prettier#cli-helper-tool) can help you remove [formatting-related rules](https://prettier.io/docs/en/comparison.html#how-does-it-compare-to-eslinttslintstylelint-etc) from your ESLint config.
